contrib/go-chi/chi/option.go

```diff
@@ -71,3 +74,15 @@ func WithAnalyticsRate(rate float64) Option {
 		}
 	}
 }
+
+// WithStatusCheck sets the function which is used to decide whther the
+// status should be marked as an error
+func WithStatusCheck(statusCheck func(statusCode int) bool) Option {
```

Comment:
```suggestion
func WithStatusCheck(fn func(statusCode int) bool) Option {
```
No need for the longer name imho.